Sources/Services/ContainerSandboxService/SandboxService.swift

```diff
@@ -225,14 +248,38 @@ public actor SandboxService {
             if id == containerId {
                 try await self.startInitProcess(lock: lock)
                 await self.setState(.running)
-                try await self.sendContainerEvent(.containerStart(id: id))
             } else {
                 try await self.startExecProcess(processId: id, lock: lock)
             }
             return message.reply()
         }
     }
 
+    /// Shutdown the SandboxService.
+    ///
+    /// - Parameters:
+    ///   - message: An XPC message with no parameters.
+    ///
+    /// - Returns: An XPC message with no parameters.
+    @Sendable
+    public func shutdown(_ message: XPCMessage) async throws -> XPCMessage {
+        self.log.info("`shutdown` xpc handler")
+
+        self.setState(.shuttingDown)
+
+        Task {
+            do {
+                try await Task.sleep(for: .seconds(3))
+            } catch {
+                self.log.error("failed to sleep before shutting down SandboxService: \(error)")
+            }
+            self.log.info("Shutting down SandboxService")
+            exit(0)
+        }
+
+        return message.reply()
+    }
+
```

Comment:
This is a nit and might be a matter of preference but perhaps we should move the private methods that follow down below `public dial()`.

Reply:
Agree